### ISSUE TITLE
Battery plugin needs "capacity" entry in types.db, default "true" value for ReportDegraded

### ISSUE THREAD

**Issue by ryan-williams** [collectd/collectd#1338]
Hi there, I just installed collectd (5.5.0 on OSX 10.11.1) via `brew` and enabled the battery plugin by uncommenting these lines that came in the conf file:

```
LoadPlugin battery
…
<Plugin "battery">
  ValuesPercentage false
  ReportDegraded
</Plugin>
```

When I run `collectd`, it exits immediately with this error:

```
Parse error in file `/usr/local/etc/collectd.conf', line 293 near `<newline>': syntax error, unexpected EOL
yyparse returned error #1
configfile: Cannot read file `/usr/local/etc/collectd.conf'.
Unable to read config file /usr/local/etc/collectd.conf.
Error: Reading the config file failed!
Read the syslog for details.
```

which points to the `ReportDegraded` line. It seems to not like the implied default `true` value, as adding that fixes it:

```
LoadPlugin battery
…
<Plugin "battery">
  ValuesPercentage false
  ReportDegraded true
</Plugin>
```

However, on restarting collectd again I get these error messages every time collectd tries to report stats (every 10s):

```
[2015-11-05 17:13:01] [info] plugin_dispatch_values: Dataset not found: capacity (from "mbp.local/battery-0/capacity-charged"), check your types.db!
[2015-11-05 17:13:01] [info] plugin_dispatch_values: Dataset not found: capacity (from "mbp.local/battery-0/capacity-discharged"), check your types.db!
[2015-11-05 17:13:01] [info] plugin_dispatch_values: Dataset not found: capacity (from "mbp.local/battery-0/capacity-degraded"), check your types.db!
```

It seems like a "capacity" definition is missing from the provided `types.db`, so I added an additional `TypesDB` directive pointing to a file with a line like:

```
capacity        value:GAUGE:0:U
```

That seems to resolve the collectd errors. I'm still unable to get influxDB to process the battery measurements, though I suspect that may be an InfluxDB problem related to the values being output in scientific notation, e.g. `9.221120237041091e+18`.

In any case, the two snags above (default "true" and "capacity" types.db) seem like things collectd {c,sh}ould fix to make the battery plugin work more seamlessly out of the box? Thanks!

**Comment by ryan-williams** [comment]
Also, it's maybe worth noting that the values I've seen output by the battery plugin, when I list `capacity` as an `ABSOLUTE` (cf. https://github.com/influxdb/influxdb/issues/4687), are all `9.221120237041091e+18`, which is suspiciously close, but not equal to, 2^63 (~2bp off); not sure if that's Influx parsing the values incorrectly or the battery plugin reporting such huge numbers for some reason.

**Comment by mfournier** [comment]
Thanks a lot for this detailed report @ryan-williams ! This will be included in collectd 5.5.1.

As for the influxdb issue, isn't it the same problem as #949 ?

**Comment by ryan-williams** [comment]
Thanks for the quick fix @mfournier!

I don't totally grok what's happening in #949→#1215; is it that the network "gauge"s are emitting cumulative / monotonically-increasing values, where they should be emitting rates per unit time?

If so, it's possible that my downstream issue here (consuming battery values in InfluxDB) is analogous to that, i.e. that the unreasonably-large numbers I'm seeing emitted above are in fact the sum of some long-running accumulator of battery values? I can't say I totally understand the hypothesis. Is it just a coincidence that the accumulator seems to currently be hovering within a few orders of magnitude of 2^63?

Also, I have one outstanding concern and one question about this patch:

1. in between me filing this and you fixing it, some dust settled over at https://github.com/influxdb/influxdb/issues/4687 and it seems like collectd was emitting NaNs when `capacity` was a `GAUGE`, and correct values when it was an `ABSOLUTE`. I'm too new to both these tools to know whether/why that might be, but AFAICT having `capacity` as a `GAUGE` per https://github.com/collectd/collectd/commit/b7de71b32b11645957e210ebadc86ca6cd84041f results in collectd emitting NaNs. I'm curious to hear your thoughts about this.
2. Is there some reason that the battery plugin was working without these modifications for others, since https://github.com/collectd/collectd/commit/a2e732f8fc1673f0f30ac93aefab32d757ea17c3 over a year ago?
